.github/workflows/docker-bazel-base-builder.yml

```diff
@@ -0,0 +1,51 @@
+---
+name: "Build Bazel base Docker image"
+on: # yamllint disable-line rule:truthy
+  push:
+    branches:
+      - master
+    paths:
+      - experimental/bazel-base/Dockerfile
+      - .github/workflows/docker-bazel-base-builder.yml
+  schedule:
+    - cron:  '13 3 * * *'
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_STREAM: ${{ github.repository }}/bazel-base
+  IMAGE_TAGS: type=sha
+  DOCKERFILE: experimental/bazel-base/Dockerfile
+
+jobs:
+  build_dockerfile:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Out Repo
+        uses: actions/checkout@v2
+      - name: Set up Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          # defines the image stream the image is pushed to
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_STREAM }}
+          # defines the image tags added to the image
+          tags: ${{ env.IMAGE_TAGS }}
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to GHCR
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.repository_owner }}
+          password:  ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push Docker image
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ${{ env.DOCKERFILE }}
+          # without docker/metadata-action will only be accessible by hash
+          push: true # will lead to authentication error when startet from PR
```

Comment:
I think this rule will not run for PRs due to:
```
  push:
    branches:
      - master
  ```
  So It will only run for pushes to master. For PRs one could use something like https://github.com/magma/magma/blob/b737e22959a49b8585ae4f25b2c871ae9a3ff8e2/.github/workflows/agw-workflow.yml#L9-L13

Reply:
To check the build validity the build was intermediatly be run on PR in https://github.com/magma/magma/commit/69520081f036ebf096d97b80f6a598c29cc7b651.